tomb

```diff
@@ -1362,10 +1362,12 @@ list_gnupg_ciphers() {
 	# prints an error if GnuPG is not found
 	which gpg &>/dev/null || _failure "gpg (GnuPG) is not found, Tomb cannot function without it."
 
+	# On gpg1 and gpg2 line 10 always point to available ciphers.
+	# Print those until the next section is found
 	ciphers=(`gpg --version | awk '
 BEGIN { ciphers=0 }
-/^Cipher:/ { gsub(/,/,""); sub(/^Cipher:/,""); print; ciphers=1; next }
-/^Hash:/ { ciphers=0 }
+NR==11 { gsub(/,/,""); sub(/^.*:/,""); print; ciphers=1; next }
```

Comment:
How can we be sure that the line listing the ciphers is always 11 in previous and future versions of GnuPG?
I think a safer bet is to assume the line is prefixed with Cipher:

Reply:
Yeah, not easy to decide what is the best.
If we go with Cipher, then we need to force locale with LANG=C, which could fail if LC_ALL or LC_COLLATE is set. And I kinda dislike such a solution, as it sets on a fixed setting (although in this case it might be forgiveable).
Then there could be some sed magic, but dunno if this will be easier.

Regarding previous and future versions:
I looked into the git history.
Two functions are important.
1.  show_version() from util/argparse.c . Comparing this function against the latest gpg1 release it was like that since the [GPL3 switch](https://git.gnupg.org/cgi-bin/gitweb.cgi?p=gnupg.git;a=blobdiff;f=util/argparse.c;h=601f6660ef3aaed01e11538514f4a2334981fe17;hp=cd29c0cef6de4e61a2b13f957cad6bb721b25abb;hb=9a2a818887b4d36ca5aed23fb0381e0e02653b6a;hpb=960ac800487dbc9d91bbef005fee4ecbfae72e83) done 2007 (so earliest gpg1 that would work: good question. Earliest visible tag is 1.4.14 from 2013. But I assume there were some versions before). What was missing in the previous implementations was the License info.

1. strusage() from g10/gpg.c. Should be working with earlier implementations as well, but the latest gpg1 matches the one from [2005](https://git.gnupg.org/cgi-bin/gitweb.cgi?p=gnupg.git;a=blob;f=g10/gpg.c;h=10eb24c413a2a4a43b31f1e01867e7e606c618fe;hb=bd1df0119c26d14e298d11abdd07ae77e8fb124f).

Tested the fixed linenumber with gpg1 and gpg 2.2.4. So, fingers crossed, I don't assume there will be changes anytime soon.

That would my arguments :D Your call :P